src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -1197,4 +1226,91 @@ ObjectDataHandler::clear_ret ObjectDataHandler::clear(
     });
 }
 
+ObjectDataHandler::clone_ret ObjectDataHandler::clone(
+  context_t ctx)
+{
+  return with_objects_data(
+    ctx,
+    [ctx, this](auto &object_data, auto &d_object_data) {
+    ceph_assert(d_object_data.is_null());
+    d_object_data.update_reserved(
+      object_data.get_reserved_data_base(),
+      object_data.get_reserved_data_len());
+    object_data.clear();
+    return prepare_data_reservation(
+      ctx,
+      object_data,
+      d_object_data.get_reserved_data_len()
+    ).si_then([&d_object_data, ctx, &object_data](auto pin) {
+      ceph_assert(pin);
+      object_data.update_reserved(pin->get_key(), pin->get_length());
```

Comment:
in prepare_data_reservation already do object_data.update_reserved,right?

Reply:
Ah, yes.